src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -175,5 +179,48 @@ private static void writeResult(String benchmarkResult, String outputFilePath, M
             writer.stepOut();
             writer.stepOut();
         }
+        String regressionDetectResult = detectRegression(thresholdFilePath, scoreMap, outputFilePath);
+        // This print out value will be passed to one environment variable in the GitHub Actions workflow.
+        System.out.println(regressionDetectResult);
+    }
+
+    /**
+     * Compare the relative changes of benchmark results with the thresholds, if the relative change smaller than threshold score which represent the decrease threshold of one
+     * specific aspect, then the performance regression detected and return 'true'.
+     * @param thresholdFilePath specifies the path of file which contains all threshold scores.
+     * @param scoreMap is a hashmap which match relative change of the score with the aspect it represents.
+     * @param outputFilePath is the destination of generated report after comparison process.
+     * @return a String which contains the information about whether performance regression happened.
+     * @throws Exception if occur happen when reading Ion Data.
+     */
+    private static String detectRegression(String thresholdFilePath, Map<String, BigDecimal> scoreMap, String outputFilePath) throws Exception {
```

Comment:
I'd like to see a unit test for this method to make sure it detects regressions when expected.